### PR TITLE
Print CL status code on error in opengl interop functions

### PR DESCRIPTION
**Pull request by kallaballa**
Pertaining Issue: #22854

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by asmorkalov — COMMENTED** [review]
I recommend you to use `CV_Error_`. It supports formatting strings like printf. See https://docs.opencv.org/4.x/db/de0/group__core__utils.html#ga1c0cd6e5bd9a5f915c6cab9c0632f969

**Comment by kallaballa** [comment]
Alright

**Comment by kallaballa** [comment]
Done [39c9027](https://github.com/opencv/opencv/pull/22855/commits/39c9027b532794145eb946caf85a088883e8bf71)

**Review by asmorkalov — APPROVED** [review]

File: modules/core/src/opengl.cpp [hunk 1]
```diff
@@ -1638,14 +1638,14 @@ Context& initializeContextFromGL()
     cl_uint numPlatforms;
     cl_int status = clGetPlatformIDs(0, NULL, &numPlatforms);
     if (status != CL_SUCCESS)
-        CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get number of platforms");
+        CV_Error_(cv::Error::OpenCLInitError, "OpenCL: Can't get number of platforms: %d ", status);
```
Comment (alalek) [inline modules/core/src/opengl.cpp:1641]: Is it tested anywhere on CI?

Used syntax is following:

`CV_Error_(core, (format, args...))`

Comment (asmorkalov) [reply modules/core/src/opengl.cpp:1641]: I got your point and fix it asap.

Comment (kallaballa) [reply modules/core/src/opengl.cpp:1641]: I am testing at the moment. Why did this work?

Comment (kallaballa) [reply modules/core/src/opengl.cpp:1641]: A merge gone bad🤦‍♂️
Still testing.

**Comment by kallaballa** [comment]
Fixed: https://github.com/kallaballa/opencv/commit/8a3696691728ec504bca007e3622f8ffa2c72e63

**Pull request merged**